### PR TITLE
Update content

### PR DESCRIPTION
**Pull request by lizbur10**
Existing lesson with substantial updates

### CODE REVIEW

**Review by linda-seiter — APPROVED** [review]
-is required for the callback function. Recall from the last lesson that, Iif
+is required for the callback function. Recall from the last lesson that, if

File: README.md [hunk 1]
```diff
@@ -2,113 +2,174 @@
...
+`Array.prototype.find()`, which can be used for a wide array of search
+conditions, and `Array.prototype.indexOf()`, which can be used to search for a
+specific value.
+
+As you work through this lesson, you will notice that there is a lot of overlap
+in how `forEach()` and `find()` work (`indexOf()` is a bit different). This will
+be the case for all of the iterator methods you will learn about in this
+module other than `indexOf()`. Specifically, they all:
+
+1. iterate through the elements in the array they're called on,
```
Comment (jlboba) [inline README.md:29]: Noticed this in the `forEach` lesson as well, but maybe for consistency in lists make these unconnected sentences instead? i.e. 

1. Iterate through the elements in the array they're called on.
2. Take a callback as an argument.
3. Automatically pass value to the callback.
4. Return a value at the end (with the exception of `forEach()`).

**Review by jlboba — COMMENTED** [review]
Looks good! Love all the checks and challenges you add throughout.

**Review by ChelseaLrn — COMMENTED** [review]
Looks great!

**Pull request merged**